packages/utils/src/object.ts

```diff
@@ -54,16 +56,31 @@ export function get(
   fallback?: any,
   index?: number,
 ) {
-  // @ts-ignore
-  path = (path?.split?.(".") ?? [path]) as string
-  for (index = 0; index < path.length; index += 1) {
-    obj = obj ? obj[path[index]] : undefined
+  const key = typeof path === "string" ? path.split(".") : [path]
+
+  if (!getCache.has(obj)) {
+    getCache.set(obj, new Map())
+  }
+
+  const map = getCache.get(obj)
+
+  if (map.has(key)) {
+    return map.get(key)
   }
-  return obj === undefined ? fallback : obj
-}
 
-// Just a memoized version of `get`
-export const memoizedGet = memoizeOne(get)
+  for (index = 0; index < key.length; index += 1) {
+    if (!obj) {
+      break
+    }
+
+    obj = obj[key[index]]
+  }
+
+  const value = obj === undefined ? fallback : obj
+  map.set(key, value)
+
+  return value
+}
```

Comment:
Just looking through this and I feel like we can split the cache part and the actual `get` function, then compose them together.

We're doing 2 things within a single function, and I'd like us to separate that.

```js
const memoGet = memoize(get)
```

Can we do this?

Reply:
should be resolved, readded memoizedGet where it was removed previously